### PR TITLE
test: add e2e test for json format

### PR DESCRIPTION
**Pull request by rst0git**
This pull request adds a test case that verifies the output JSON format for checkpointctl.

### CODE REVIEW

**Comment by rst0git** [comment]
cc: @Parthiba-Hazra

**Review by snprajwal — APPROVED** [review]

**Comment by adrianreber** [comment]
I see a coverage of 20% of `json.go`. Would a higher coverage require a more complex test process?

**Comment by rst0git** [comment]
> Would a higher coverage require a more complex test process?

Not necessarily. I will update the patch to extend the test coverage.

**Comment by rst0git** [comment]
@adrianreber I've updated the pull request to increase test coverage.

File: test/checkpointctl.bats [hunk 1]
```diff
@@ -14,7 +14,15 @@ function checkpointctl() {
...
 }
 
+run_only_test() {
+	if [ "$BATS_TEST_NUMBER" -ne "$1" ]; then
+		skip
+	fi
+}
+
 function setup() {
+	# run_only_test 49
```
Comment (adrianreber) [inline test/checkpointctl.bats:24]: Is this to only run certain tests? I was always using

`-f, --filter <regex>      Only run tests that match the regular expression`

To only run one test.

File: test/piggie/piggie.c [hunk 2]
```diff
@@ -34,6 +144,9 @@ static int do_test(void *logf)
 	if (fd != 1 && fd != 2)
 		close(fd);
 
+	run_tcp_server();
```
Comment (adrianreber) [inline test/piggie/piggie.c:147]: Maybe we could only run the TCP part for selected tests. I am bit worried that always opening port 5000 will result in some problems sometimes. Port already in use or something. Alternatively you could try to open the port in a network namespace which would also include namespaces in the test binary.

File: test/checkpointctl.bats (hunk 1 above)
Comment (rst0git) [reply test/checkpointctl.bats]: Thanks! I've removed this function.

File: test/piggie/piggie.c (hunk 2 above)
Comment (rst0git) [reply test/piggie/piggie.c]: Good point! I've updated the test to create a network namepsace and use command-line options for TCP sockets and log file.

**Review by adrianreber — APPROVED** [review]

**Pull request merged**